Validation/Geometry/src/MaterialBudgetData.cc

```diff
@@ -20,217 +16,39 @@ MaterialBudgetData::MaterialBudgetData()
-  theFinalY   = new double[MAXNUMBERSTEPS];
-  theFinalZ   = new double[MAXNUMBERSTEPS];
-  // rr
-  theVolumeID     = new int[MAXNUMBERSTEPS];
-  theVolumeName   = new std::string[MAXNUMBERSTEPS];
-  theVolumeCopy   = new int[MAXNUMBERSTEPS];
-  theVolumeX      = new float[MAXNUMBERSTEPS];
-  theVolumeY      = new float[MAXNUMBERSTEPS];
-  theVolumeZ      = new float[MAXNUMBERSTEPS];
-  theVolumeXaxis1 = new float[MAXNUMBERSTEPS];
-  theVolumeXaxis2 = new float[MAXNUMBERSTEPS];
-  theVolumeXaxis3 = new float[MAXNUMBERSTEPS];
-  theVolumeYaxis1 = new float[MAXNUMBERSTEPS];
-  theVolumeYaxis2 = new float[MAXNUMBERSTEPS];
-  theVolumeYaxis3 = new float[MAXNUMBERSTEPS];
-  theVolumeZaxis1 = new float[MAXNUMBERSTEPS];
-  theVolumeZaxis2 = new float[MAXNUMBERSTEPS];
-  theVolumeZaxis3 = new float[MAXNUMBERSTEPS];
-  theMaterialID      = new int[MAXNUMBERSTEPS];
-  theMaterialName    = new std::string[MAXNUMBERSTEPS];
-  theMaterialX0      = new float[MAXNUMBERSTEPS];
-  theMaterialLambda0 = new float[MAXNUMBERSTEPS];
-  theMaterialDensity = new float[MAXNUMBERSTEPS];
-  theStepID             = new int[MAXNUMBERSTEPS];
-  theStepInitialPt      = new float[MAXNUMBERSTEPS];
-  theStepInitialEta     = new float[MAXNUMBERSTEPS];
-  theStepInitialPhi     = new float[MAXNUMBERSTEPS];
-  theStepInitialEnergy  = new float[MAXNUMBERSTEPS];
-  theStepInitialPx      = new float[MAXNUMBERSTEPS];
-  theStepInitialPy      = new float[MAXNUMBERSTEPS];
-  theStepInitialPz      = new float[MAXNUMBERSTEPS];
-  theStepInitialBeta    = new float[MAXNUMBERSTEPS];
-  theStepInitialGamma   = new float[MAXNUMBERSTEPS];
-  theStepInitialMass    = new float[MAXNUMBERSTEPS];
-  theStepFinalPt        = new float[MAXNUMBERSTEPS];
-  theStepFinalEta       = new float[MAXNUMBERSTEPS];
-  theStepFinalPhi       = new float[MAXNUMBERSTEPS];
-  theStepFinalEnergy    = new float[MAXNUMBERSTEPS];
-  theStepFinalPx        = new float[MAXNUMBERSTEPS];
-  theStepFinalPy        = new float[MAXNUMBERSTEPS];
-  theStepFinalPz        = new float[MAXNUMBERSTEPS];
-  theStepFinalBeta      = new float[MAXNUMBERSTEPS];
-  theStepFinalGamma     = new float[MAXNUMBERSTEPS];
-  theStepFinalMass      = new float[MAXNUMBERSTEPS];
-  theStepPreProcess     = new int[MAXNUMBERSTEPS];
-  theStepPostProcess    = new int[MAXNUMBERSTEPS];
-  // rr
 }
 
 
 void MaterialBudgetData::dataStartTrack( const G4Track* aTrack )
 {
-  std::cout << "MaterialBudgetData::dataStartTrack( const G4Track* aTrack )" << std::endl;;
+
   const G4ThreeVector& dir = aTrack->GetMomentum() ;
 
-  if( myMaterialBudgetCategorizer == nullptr) myMaterialBudgetCategorizer = new MaterialBudgetCategorizer;
+  if( myMaterialBudgetCategorizer == nullptr){
+    myMaterialBudgetCategorizer = std::make_unique<MaterialBudgetCategorizer>();
+  }
   
   theStepN=0;
   theTotalMB=0;
   theTotalIL=0;
   theEta=0;
   thePhi=0;
-  
-  // rr
   theID=0;
   thePt=0;
   theEnergy=0;
   theMass=0;
-  
+ 
   theSupportMB     = 0.;
```

Comment:
@vargasa - to avoid a conversion from double to float, please use ```0.f```
```theSupportMB``` is float, ```0.``` is double:
https://en.cppreference.com/w/cpp/language/floating_literal